src/figuresClasses.ts

```
@@ -1,11 +1,88 @@
-export interface Figure {}
+type СolorsType = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string;
+
+  constructor(
+    public color: СolorsType,
```

This violates checklist item: "Triangle constructor: No validation that `color` is one of `red`, `green`, `blue`." Although the constructor parameter uses the `СolorsType` alias, there is no runtime validation that the passed value is one of the allowed colors. Add a runtime check (for example `const allowed = ['red','green','blue']` and `if (!allowed.includes(this.color)) throw new Error("Color must be one of: 'red', 'green', 'blue'")`) to meet the requirement and fail fast if an invalid string is passed at runtime.